### PR TITLE
fix: race condition in goreleaser workflow

### PR DESCRIPTION
**Pull request by erikburt**
### Changes

- Use the `github.sha` rather than `github.ref` during goreleaser build/merges to fix race condition

### Motivation

On a push event:
1. Goreleaser workflow is triggered
2. split job is done first: 
	1. This builds all amd64 and arm64 images through a build matrix
3. Then the merge job is waiting for split to finish
    1. merge checks out the repo at `github.ref`

If there was a push to develop in between the initial dispatch and the merge starting, the expected SHA will differ, and the version check will fail.

---

RE-3215

### CODE REVIEW

**Comment by erikburt** [comment]
Other solution is to set a concurrency group and cancel in-progress runs if a new commit has been pushed. Thoughts?

File: .github/workflows/build-publish-develop-pr.yml [hunk 1]
```diff
@@ -24,7 +24,10 @@ on:
         default: "false"
 
 env:
-  GIT_REF: ${{ github.event.inputs.git_ref || github.ref }}
+  # Use github.sha here otherwise a race condition exists if
+  # a commit is pushed to develop besfore merge is run.
```
Comment (chainchad) [inline .github/workflows/build-publish-develop-pr.yml]:
```suggestion
  # a commit is pushed to develop before merge is run.
```

File: .github/workflows/build-publish-develop-pr.yml [hunk 2]
```diff
@@ -103,7 +106,7 @@ jobs:
           role-session-name: "split-${{ matrix.goarch }}"
 
       - id: cache
-        uses: actions/cache@v4
+        uses: actions/cache/@v4.1.1
```
Comment (chainchad) [inline .github/workflows/build-publish-develop-pr.yml]:
```suggestion
        uses: actions/cache@v4.1.1
```

**Comment by chainchad** [comment]
Looks good, nice find!

File: .github/workflows/build-publish-develop-pr.yml (hunk 2 above)
Comment (erikburt) [reply .github/workflows/build-publish-develop-pr.yml]: 5997cce1516021900c717c26a240269789919c33

File: .github/workflows/build-publish-develop-pr.yml (hunk 1 above)
Comment (erikburt) [reply .github/workflows/build-publish-develop-pr.yml]: 5997cce1516021900c717c26a240269789919c33

**Review by chainchad — APPROVED** [review]

**Review by momentmaker — APPROVED** [review]

**Pull request merged**